power-of-thor.lua

```diff
@@ -0,0 +1,45 @@
+next_token = string.gmatch(io.read(), "[^%s]+")
+LX = tonumber(next_token()) -- Position initiale du personnage
+LY = tonumber(next_token())
+TX = tonumber(next_token()) -- Position de la cible
+TY = tonumber(next_token())
+
+-- Boucle du jeu
+while true do
+    remainingTurns = tonumber(io.read())
+
+    -- Calcul de la direction
+    local direction = ""
+
+    if LY > TY then
+        direction = direction .. "S"
+        LY = LY - 1
+    elseif LY < TY then
+        direction = direction .. "N"
+        LY = LY + 1
+    end
+
+    if LX > TX then
+        direction = direction .. "E"
+        LX = LX - 1
+    elseif LX < TX then
+        direction = direction .. "W"
+        LX = LX + 1
+    end
+
+    -- Afficher la direction du tour
+    print(direction)
+
+    -- Calcul de la distance (hypoténuse)
+    local absX = math.abs(LX - TX)
+    local absY = math.abs(LY - TY)
+    local distance = math.sqrt(absX^2 + absY^2)
```

Comment:
comme tu compares distance à 0, tu peux simplifier le calcul

par exemple :
```suggestion
    local distance = absX^2 + absY^2
```

mais tu peux faire mieux. Je te laisse chercher